test/telemetry_metrics_appsignal_test.exs

```diff
@@ -35,39 +45,19 @@ defmodule TelemetryMetricsAppsignalTest do
       [:db, :query] => [Summary]
     }
 
-    Enum.each(event_metrics, fn {event_name, expected_metrics} ->
-      handler_id = [@handler_prefix | event_name] |> Enum.join("_")
-      handler = Enum.find(attached_handlers, &(&1.id == handler_id))
-      handler_metrics = handler.config[:metrics]
-
-      assert handler.event_name == event_name
-      assert Enum.map(handler_metrics, & &1.__struct__()) == expected_metrics
-    end)
+    actual_event_metrics =
+      Enum.reduce(attached_handlers, %{}, fn handler, metrics_acc ->
+        handler_metrics = handler.config[:metrics]
+        event_name = List.first(handler.config[:metrics]).event_name
+        modules = Enum.map(handler_metrics, & &1.__struct__())
+        Map.put(metrics_acc, event_name, modules)
+      end)
 
-    TelemetryMetricsAppsignal.detach(metrics)
-    attached_handlers = :telemetry.list_handlers([])
-    assert attached_handlers == []
-  end
+    assert Enum.sort(actual_event_metrics) == Enum.sort(event_metrics)
 
-  test "allwing to configure handler namespace" do
-    metric = counter("web.request.count")
-    TelemetryMetricsAppsignal.attach([metric], namespace: "api")
-    TelemetryMetricsAppsignal.attach([metric], namespace: "rpc")
+    Agent.stop(reporter)
```

Comment:
Shouldn't we use [`ExUnit.Callbacks.stop_supervised!/1`](https://hexdocs.pm/ex_unit/ExUnit.Callbacks.html#stop_supervised!/1)?

Reply:
That makes much more sense.